### PR TITLE
Add URL post link

### LINKED ISSUE #145: URL in the post should be linked automatically
image

### PR DESCRIPTION
**Pull request by Zuubar**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #145

### CODE REVIEW

File: modules/v1/workspaces/workspaceBehaviours/TimelineUrlBehaviour.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        return [
+            ActiveRecord::EVENT_AFTER_INSERT => 'afterInsert',
+            ActiveRecord::EVENT_AFTER_UPDATE => 'afterInsert',
+        ];
+    }
+
+    public function afterInsert() {
+        if(strlen($this->timelinePost->description !== 0)) {
+            $description = $this->timelinePost->description;
+            $description = strip_tags(preg_replace("((<\/)\w+(>))", "\n", $description));
```
Comment (thecodeholic) [inline modules/v1/workspaces/workspaceBehaviours/TimelineUrlBehaviour.php]: Can you explain what this does?

File: modules/v1/workspaces/workspaceBehaviours/TimelineUrlBehaviour.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            $descriptions = preg_split("/[\n\s,]+/", $description);
```
Comment (thecodeholic) [inline modules/v1/workspaces/workspaceBehaviours/TimelineUrlBehaviour.php]: or this?

File: modules/v1/workspaces/workspaceBehaviours/TimelineUrlBehaviour.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+            $regex = "/^(http|https):\/\/[^\s]+/";
+            foreach ($descriptions as $i) {
```
Comment (thecodeholic) [inline modules/v1/workspaces/workspaceBehaviours/TimelineUrlBehaviour.php]: You are iterating char by char in the description which obviously is not necessary.
You just need to write preg_replace once.

File: modules/v1/workspaces/models/TimelinePost.php [hunk 4]
```diff
@@ -61,6 +62,10 @@ public function behaviors()
         return array_merge(parent::behaviors(), [
             TimestampBehavior::class,
             BlameableBehavior::class,
+            [
+                'class' => TimelineUrlBehaviour::class,
+                'timelinePost' => $this
```
Comment (thecodeholic) [inline modules/v1/workspaces/models/TimelinePost.php]: In this specific case there is no need to create behavior, you can even write necessary code in model, but if you want to create universal behavior, better will be if you pass property name(s) as an array, but not $this.

**Comment by thecodeholic** [comment]
@Null-IO Let me know when you have time and we should talk about this task.

File: modules/v1/workspaces/workspaceBehaviours/TimelineUrlBehaviour.php (hunk 1 above)
Comment (Zuubar) [reply modules/v1/workspaces/workspaceBehaviours/TimelineUrlBehaviour.php]: Basically, description in db is kept with html format `<p>description</p>`. I strip tags from description and apply link regex on "clean description"

File: modules/v1/workspaces/workspaceBehaviours/TimelineUrlBehaviour.php (hunk 2 above)
Comment (Zuubar) [reply modules/v1/workspaces/workspaceBehaviours/TimelineUrlBehaviour.php]: Same reason as above

File: modules/v1/workspaces/workspaceBehaviours/TimelineUrlBehaviour.php (hunk 3 above)
Comment (Zuubar) [reply modules/v1/workspaces/workspaceBehaviours/TimelineUrlBehaviour.php]: With case of multiline description, tags are adding up `<p>description</p><p>newline</p>.` After splitting, it becomes an array of "clean descriptions" and I iterate over them and apply link regex

File: modules/v1/workspaces/models/TimelinePost.php (hunk 4 above)
Comment (Zuubar) [reply modules/v1/workspaces/models/TimelinePost.php]: This behaviour was created specifically for timeline posts. If same kind of behaviour is needed for other models, i will look into it

**Pull request merged**